packages/tools/examples/logicalOperators/index.ts

```diff
@@ -0,0 +1,332 @@
+  segmentation.segmentIndex.setActiveSegmentIndex(segmentationId, segmentIndex);
+}
+
+function getSegmentsVisibilityState() {
+  let segmentsVisibility = segmentVisibilityMap.get(segmentationId);
+
+  if (!segmentsVisibility) {
+    segmentsVisibility = new Array(segmentIndexes.length + 1).fill(true);
+    segmentVisibilityMap.set(segmentationId, segmentsVisibility);
+  }
+
+  return segmentsVisibility;
+}
+
+function updateSegmentationConfig(config) {
+  segmentation.config.style.setStyle(
+    {
+      segmentationId,
+      type: csToolsEnums.SegmentationRepresentations.Contour,
+    },
+    config
+  );
+}
+
+// ============================= //
+
+const cancelDrawingEventListener = (evt) => {
+  const { element, key } = evt.detail;
+  if (key === 'Escape') {
+    cornerstoneTools.cancelActiveManipulations(element);
+  }
+};
+
+element.addEventListener(
+  csToolsEnums.Events.KEY_DOWN,
+  cancelDrawingEventListener
+);
+
+const Splines = {
+  CatmullRomSplineROI: {
+    splineType: SplineContourSegmentationTool.SplineTypes.CatmullRom,
+  },
+  LinearSplineROI: {
+    splineType: SplineContourSegmentationTool.SplineTypes.Linear,
+  },
+  BSplineROI: {
+    splineType: SplineContourSegmentationTool.SplineTypes.BSpline,
+  },
+};
+
+const SplineToolNames = Object.keys(Splines);
+const splineToolsNames = [...SplineToolNames];
+let selectedToolName = splineToolsNames[0];
+
+addDropdownToToolbar({
+  labelText: 'Segment Index',
+  options: { values: segmentIndexes, defaultValue: segmentIndexes[0] },
+  onSelectedValueChange: (nameAsStringOrNumber) => {
+    updateActiveSegmentIndex(Number(nameAsStringOrNumber));
+  },
+});
+
+addDropdownToToolbar({
+  options: { values: splineToolsNames, defaultValue: selectedToolName },
+  onSelectedValueChange: (newSelectedToolNameAsStringOrNumber) => {
+    const newSelectedToolName = String(newSelectedToolNameAsStringOrNumber);
+    const toolGroup = ToolGroupManager.getToolGroup(toolGroupId);
+
+    // Set the old tool passive
+    toolGroup.setToolPassive(selectedToolName);
+
+    // Set the new tool active
+    toolGroup.setToolActive(newSelectedToolName, {
+      bindings: contourSegmentationToolBindings,
+    });
+
+    selectedToolName = <string>newSelectedToolName;
+  },
+});
+
+function performLogicalOperation(operation: number = 1) {
+  const activeSeg = segmentation.getActiveSegmentation(viewportId);
+  const renderEngine = getRenderingEngine(renderingEngineId);
+  const viewport = renderEngine.getViewport(viewportId);
+
+  if (!activeSeg) {
+    console.log('No active segmentation detected');
+    return;
+  }
+
+  if (!activeSeg.representationData.Contour) {
+    console.log('No contour representation found');
+    return;
+  }
+
+  const representationData = activeSeg.representationData.Contour;
+  const { annotationUIDsMap } = representationData;
+
+  const { addition, subtraction } =
+    cornerstoneTools.utilities.contourSegmentation;
+  if (annotationUIDsMap) {
```

Comment:
the usage should be something like `cornerstoneTools.segmentation.operators.add({ segmentationId: string, segmentIndex: integer}, {segmentationId: string, segmentIndex: integer}, options)` 
so two operands

There is no need for viewport or active seg, 
it is done on segmentationId, not the representation (throw error if the segmentation does not have a representation of contour)